docs/solidity-by-example.rst

```diff
@@ -645,4 +645,499 @@ Safe Remote Purchase
 Micropayment Channel
 ********************
 
-To be written.
+Creating and verifying signatures
+=================================
+
+Signatures are used to authorize transactions,
+and they're a general tool that's available to
+smart contracts. In this chapter we'll use it to
+prove to a smart contract that a certain account
+approved a certain message. We'll build a simple
+smart contract that lets me transmit ether, but
+in a unsual way, instead of calling a function myself
+to initiate a payment, I'll let the receiver of
+the payment do that, and therefore pay the transaction
+fee. All this using cryptographics signatures that I'll
+make and send off-chain (e.g. via email). We'll call our
+contract as ReceiverPays, it will works a lot like
+writing a check.Our contract will works like that:
```

Comment:
Perhaps it would be better to mention the check analogy first and then describe the more abstract mechanism of who initiates the transaction?